### PR TITLE
gpinitsystem -p does not produce correct postgresql.conf file

### PR DESCRIPTION
**Pull request by eurekaka**
when the add-on config file contains GUCs which exist in postgresql.conf.sample,
and are commented out(e.g, #max_resource_queues), these GUC settings would not
take effect because incorrect posgresql.conf would be generated by gpinitsystem.

### CODE REVIEW

File: gpMgmt/bin/lib/gp_bash_functions.sh [hunk 1]
```diff
@@ -463,19 +463,29 @@ SED_PG_CONF () {
...
 			else
-                if [ $KEEP_PREV -eq 0 ];then
-                    $SED -i'.bak' -e "s/${SEARCH_TXT}/${SUB_TXT} #/g" $FILENAME
-                else
-                    $SED -i'.bak' -e "s/^${SEARCH_TXT}.*/${SUB_TXT}/g" $FILENAME
-                fi
+				if [ $KEEP_PREV -eq 0 ];then
+					$SED -i'.bak1' -e "s/${SEARCH_TXT}/${SUB_TXT} #${SEARCH_TXT}/g" $FILENAME
+				else
+					$SED -i'.bak1' -e "s/${SEARCH_TXT}.*/${SUB_TXT}/g" $FILENAME
```
Comment (danielgustafsson) [inline gpMgmt/bin/lib/gp_bash_functions.sh]: Nitpicking perhaps but can we name these files with something more illustrative than `.bak1` and `.bak2`? In case interrupted processing leaves files around it helps to identify why/what now that there are multiple files.

File: gpMgmt/bin/lib/gp_bash_functions.sh [hunk 2]
```diff
@@ -463,19 +463,29 @@ SED_PG_CONF () {
...
+				RETVAL=$?
+				if [ $RETVAL -ne 0 ]; then
+					LOG_MSG "[WARN]:-Failed to replace #$SEARCH_TXT in $FILENAME"
+					ERROR_EXIT=1
+				else
+					LOG_MSG "[INFO]:-Replaced line in $FILENAME"
+					$RM -f ${FILENAME}.bak2
 				fi
-		    fi
+			fi
```
Comment (danielgustafsson) [inline gpMgmt/bin/lib/gp_bash_functions.sh]: IMHO it helps review and diff reading when whitespace and style cleanups are done in a separate commit(s) from functionality changes.

**Comment by yydzero** [comment]
LGTM

**Comment by eurekaka** [comment]
pushed.
